mhs/inbound/inbound/request/handler.py

```diff
@@ -104,17 +107,42 @@ def _extract_message_id(self, message):
         :param message:
         :return:
         """
-        message_id = message.message_dictionary[ebxml_envelope.MESSAGE_ID]
+        message_id = message.message_dictionary[MESSAGE_ID]
         log.inbound_message_id.set(message_id)
         logger.info('009', 'Found inbound message id on request.')
 
-    def extract_ref_message(self, message):
+    def _extract_ref_message(self, message):
         """
         Extracts the reference-to message id and assigns it as the message Id in logging
         :param message:
         :return: the message id the inbound message is a response to
         """
-        message_id = message.message_dictionary[ebxml_envelope.RECEIVED_MESSAGE_ID]
+        message_id = message.message_dictionary[RECEIVED_MESSAGE_ID]
         log.message_id.set(message_id)
         logger.info('010', 'Found message id on inbound message.')
         return message_id
+
+    def _extract_incoming_async_request_message(self):
+        try:
+            request_message = ebxml_request_envelope.EbxmlRequestEnvelope.from_string(self.request.headers,
+                                                                                      self.request.body.decode())
+        except ebxml_envelope.EbXmlParsingError as e:
+            logger.error('020', 'Failed to parse response: {exception}', {'exception': e})
+            raise tornado.web.HTTPError(500, 'Error occurred during message parsing',
+                                        reason=f'Exception during inbound message parsing {e}') from e
+
+        return request_message
+
+    def _extract_incoming_sync_request_message(self):
+        try:
+            request_message = SoapEnvelope.from_string(self.request.headers,
+                                                                     self.request.body.decode())
+        except SoapParsingError as e:
+            logger.error('020', 'Failed to parse response: {exception}', {'exception': e})
```

Comment:
The log code here, '020' overlaps with the one on line 130

Reply:
Updated